### PR TITLE
Update __setatt__ content

### PR DESCRIPTION
**Pull request by Felonious-Spellfire**
## What does this PR do?

Adds note to better explain __setatt_ usage.

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [X] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [X] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source-app/core_api/lightning_app/dynamic_work_content.rst [hunk 1]
```diff
@@ -46,6 +46,25 @@ There are a couple of ways you can add a dynamic Work:
...
             getattr(self,  "work").run()
 
+.. note:: 
+     ``__setattr__()`` has to be triggered in order for the state to be replicated.
+
+     Currently a state update is triggered if the ``__setattr__()`` function is called. For example, if you use ``status.run_script = True``,
+     then ``__setattr__()`` is called and the status is updated with all found changes.
+     But if you just use ``state.form_values["script_dir"]  = st_script_dir`` then ``__setattr__()`` is not called
+     so you only get attributes and assign values to the value retrieved (not the state).
+     So that assignemnt would not update the state and would not be visible in other components.
```
Comment (awaelchli) [inline docs/source-app/core_api/lightning_app/dynamic_work_content.rst:56]: Where is this information from? This has been addressed a long time ago [1, 2] (pre-launch) by me and Thomas. If there is evidence of a regression, we need to fix it instead. 

[1] https://linear.app/lightning-ai/issue/LAI2-8436/work-state-shared-stated-doesnt-get-updated-for-in-place-operations
[2] https://github.com/Lightning-AI/lightning-app/pull/610

**Review by awaelchli — CHANGES_REQUESTED** [review]

File: docs/source-app/core_api/lightning_app/dynamic_work_content.rst (hunk 1 above)
Comment (Felonious-Spellfire) [reply docs/source-app/core_api/lightning_app/dynamic_work_content.rst:56]: @awaelchli 
Looks like this was reopened yesterday: https://linear.app/lightning-ai/issue/LAI2-9676/app-stops-with-error-in-lightningwork-attributeerror-list-object-has

Comment (awaelchli) [reply docs/source-app/core_api/lightning_app/dynamic_work_content.rst:56]: Thanks. Will wait for the reply on the linear ticket :)

**Comment by Felonious-Spellfire** [comment]
Closing this based on @awaelchli comments. This issue was resolved in a previous ticket.

**Pull request closed without merging**